### PR TITLE
Onstop issue1615

### PR DESCRIPTION
**Pull request by shwethanidd**
Added signal handler to handle SIGINT signal. Handler sets stop event which enables a clean agent shutdown.
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/kmonson%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/2049915%3Fv%3D4%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1616%23issuecomment-367865182%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222018-02-23T00%3A04%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/2049915%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kmonson%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40shwethanidd%20We%20need%20to%20look%20at%20any%20effect%20this%20may%20have%20on%20the%20platform%20executable%20at%20it%27s%20running%204%20or%205%20agents%20internally.%22%2C%20%22created_at%22%3A%20%222018-02-23T01%3A26%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/2049915%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kmonson%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20930567768bb27823e22909d6fd672d70f391aa57%20volttron/platform/vip/agent/core.py%2014%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1616%23discussion_r170128934%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%20%60%60%23%20To%20avoid%20a%20child%20agent%20handler%20overwriting%20the%20parent%20agent%20handler%60%60%22%2C%20%22created_at%22%3A%20%222018-02-22T23%3A42%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/2049915%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kmonson%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222018-02-23T00%3A04%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/2049915%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kmonson%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20volttron/platform/vip/agent/core.py%3AL170-181%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/VOLTTRON/volttron/pull/1616%23issuecomment-367865182%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1616%23discussion_r170128934%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1616%23discussion_r170132325%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1616%23issuecomment-367880175%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/kmonson'></a>

- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1616#issuecomment-367865182'>General Comment</a></b>
- <a href='https://github.com/kmonson'></a> :shipit:
- <a href='https://github.com/kmonson'></a> @shwethanidd We need to look at any effect this may have on the platform executable at it's running 4 or 5 agents internally.
- [x] <a href='#crh-comment-Pull 930567768bb27823e22909d6fd672d70f391aa57 volttron/platform/vip/agent/core.py 14'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1616#discussion_r170128934'>File: volttron/platform/vip/agent/core.py:L170-181</a></b>
- <a href='https://github.com/kmonson'></a> ``# To avoid a child agent handler overwriting the parent agent handler``

<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1616?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1616?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1616?approve=0'></a>&nbsp;<a href='https://github.com/VOLTTRON/volttron/pull/1616'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by craig8** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/VOLTTRON/volttron/pull/1616'>Refresh</a>.<a href='#crh-update-none'></a>

File: volttron/platform/vip/agent/core.py [hunk 1]
```diff
@@ -169,6 +170,11 @@ def __init__(self, owner):
         self.onstart = Signal()
         self.onstop = Signal()
         self.onfinish = Signal()
+        self.oninterrupt = None
+        prev_int_signal = gevent.signal.getsignal(signal.SIGINT)
+        # To avoid multiple handlers between parent and child agents
```
Comment (kmonson) [inline volttron/platform/vip/agent/core.py]: # To avoid a child agent handler overwriting the parent agent handler

Comment (kmonson) [inline volttron/platform/vip/agent/core.py]: :+1:

**Comment by kmonson** [comment]
:shipit:

**Comment by kmonson** [comment]
@shwethanidd We need to look at any effect this may have on the platform executable at it's running 4 or 5 agents internally.

**Pull request merged**